native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/NativeImageExecutableLocator.java

```diff
@@ -91,29 +91,29 @@ public static File findNativeImageExecutable(Property<JavaLauncher> javaLauncher
             executablePath = metadata.getInstallationPath().file("bin/" + NATIVE_IMAGE_EXE).getAsFile();
         }
 
-        try {
-            if (!executablePath.exists()) {
-                logger.log("Native Image executable wasn't found. We will now try to download it. ");
-                File graalVmHomeGuess = executablePath.getParentFile();
-
-                File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
-                if (!guPath.exists()) {
-                    throw new GradleException("'" + GU_EXE + "' at '" + guPath + "' tool wasn't found. This probably means that JDK at isn't a GraalVM distribution.");
-                }
-                ExecResult res = execOperations.exec(spec -> {
-                    spec.args("install", "native-image");
-                    spec.setExecutable(Paths.get(graalVmHomeGuess.getAbsolutePath(), GU_EXE));
-                });
-                if (res.getExitValue() != 0) {
-                    throw new GradleException("Native Image executable wasn't found, and '" + GU_EXE + "' tool failed to install it.");
-                }
-                diagnostics.withGuInstall();
+        File graalVmHomeGuess = executablePath.getParentFile();
+        File guPath = graalVmHomeGuess.toPath().resolve(GU_EXE).toFile();
```

Comment:
What if `executablePath` doesn't exist at this point? Then `getParentFile` returns null, right? Therefor, `graalVmHomeGuess` will be null and this can cause an exception?

Reply:
`File#getParentFile` does not care whether the file exists.
It only returns null if `executablePath` is `/`, which is not possible given how `executablePath` is constructed (it ends with `.../bin/native-image*` so the getParentFile is `.../bin`).